### DISCUSSION TITLE
Velero backup partially failed if I included PV and PVC.

### DISCUSSION THREAD

**Discussion by shyjustack** [vmware-tanzu/velero#6418] [Community support Q&A] [answered]
When running Velero v1.11.0 with my backup location set as S3-compatible storage, I encountered a failure when attempting to perform a full namespace backup that includes both the persistent volume (PV) and persistent volume claim (PVC). However, regular pod backups are functioning correctly.

logs 
root@HODEV-K8S-MASTER01:~# velero backup describe test-nginx-b4
Name:         test-nginx-b4
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.27.2
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=27

Phase:  PartiallyFailed (run `velero backup logs test-nginx-b4` for more information)

Namespaces:
  Included:  shyju
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto

Label selector:  <none>

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  720h0m0s

CSISnapshotTimeout:    10m0s
ItemOperationTimeout:  1h0m0s

Hooks:  <none>

Backup Format Version:  1.1.0

Started:    2023-06-22 04:51:03 +0000 UTC
Completed:  2023-06-22 04:51:07 +0000 UTC

Expiration:  2023-07-22 04:51:03 +0000 UTC

Velero-Native Snapshots: <none included>

**Comment by draghuram** [comment]
Check for error messages in Velero logs (``velero backup logs <BACKUP<``).

**Comment by shyjustack** [comment]
[root@HODEV-K8S-MASTER01~# velero ba.txt](https://github.com/vmware-tanzu/velero/files/11835791/root%40HODEV-K8S-MASTER01.velero.ba.txt)
log file attached

**Comment by draghuram** [comment]
You can see messages such as these in the log:

time="2023-06-22T15:32:29Z" level=warning msg="Failed to discover group: vault.kio.kasten.io/v1alpha1" backup=velero/demo2 cmd=/velero error="the server is currently unable to handle the request" logSource="pkg/discovery/helper.go:220" pluginName=velero

I think they contributed to "Partial" status of the backup.

**Reply by sseago** [reply, marked as answer]
Although the above discovery line is just a warning. PartiallyFailed means that there was at least one error -- look for "level=error" in the backup logs. Looks like this was the culprit:
```
time="2023-06-22T15:32:33Z" level=error msg="Error backing up item" backup=velero/demo2 error="daemonset pod not found in running state in node hodev-k8s-worker01" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/nodeagent/node_agent.go:74" error.function=github.com/vmware-tanzu/velero/pkg/nodeagent.IsRunningInNode logSource="pkg/backup/backup.go:435" name=web-0
```
This means that the node that the pod mounting the PVC being backed up by restic (or kopia) doesn't have a running node-agent pod. What is the state of the node agent daemonset? Is it in an error state? Is it only failing to run in that one node, or is it not running on any nodes?

**Reply by shyjustack** [reply]
Pods are running 
To upgrade Velero from version 9 to version 10 or 11, do we need to install any additional packages?. If you downgrade back to version 9, it iswork fine. In version 9, it seems that persistent volumes (PVs) are not being created automatically, but backups are still successfully created.
root@g42k8s-master01:~# kubectl get all -n velero
NAME                          READY   STATUS    RESTARTS   AGE
pod/velero-6dcb8c44c6-44dc4   1/1     Running   0          13d

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/velero   1/1     1            1           13d

NAME                                DESIRED   CURRENT   READY   AGE
replicaset.apps/velero-6dcb8c44c6   1         1         1       13d

**Reply by shyjustack** [reply]
hi @sseago ,

Issue: The resolve node agent was not running on the worket node. Thank you so much for your support.
root@HODEV-K8S-MASTER01:~# kubectl get all -n velero
NAME                         READY   STATUS    RESTARTS   AGE
pod/node-agent-d4wsd         1/1     Running   0          39m
pod/velero-d984cdcc4-4tcpj   1/1     Running   0          85m

NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/node-agent   1         1         1       1            1           <none>          39m

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/velero   1/1     1            1           85m

NAME                               DESIRED   CURRENT   READY   AGE
replicaset.apps/velero-d984cdcc4   1         1         1       85m

**Comment by shyjustack** [comment]
[New Text Document.txt](https://github.com/vmware-tanzu/velero/files/11846545/New.Text.Document.txt)
POD log atatched

**Comment by ShubhamGorlewar1** [comment]
I managed to solve this issue @shyju102

**Reply by shyjustack** [reply]
can you share the steps @ShubhamGorlewar1
